libraries/helpers.rb

```diff
@@ -30,6 +30,37 @@ def keys_from_url(url)
       end
     end
 
+    # Determine of a shell exists on disk
+    #
+    # @return [Boolean]
+    def shell_exists(shell_path)
+      File.exist?(shell_path)
+    end
+
+    # Determines if the user's shell is valid on the machine, otherwise
+    # returns the default of /bin/sh
+    #
+    # @return [String]
+    def default_shell(shell)
+      # A check to see if the shell exists on the system
+      unless shell_exists(shell)
+        log "Shell #{shell} not found - defaulting to /bin/sh"
+        return '/bin/sh'
+      end
+
+      return shell unless platform_family?('aix')
+
+      File.open('/etc/security/login.cfg') do |f|
```

Comment:
i think it might be best to wrap this in a `begin/rescue/end` block to be on the safe side in case `File.open` throws an error. sorry, i updated my earlier comment to reflect that but you probably missed it.

```
      begin
        File.open('/etc/security/login.cfg') do |f|
          f.each_line do |l|
            l.match(/^\s*shells\s*=\s*(.*)/) do |m|
              return shell if m[1].split(/\s*,\s*/).any? { |entry| entry.eql? shell }
            end
          end
        end
      rescue
      end
```

Reply:
What should we do if it does throw an error? (cookstyle won't let you just eat the exception)